assets/activation-scripts/activate.d/start.bash

```diff
@@ -43,6 +43,26 @@ if [ -f "$FLOX_ENV/activate.d/envrc" ]; then
   source "$FLOX_ENV/activate.d/envrc"
 fi
 
+# Start the watchdog if invoked by `flox activate` but not when the
+# `${FLOX_ENV}/activate` is invoked directly such as:
+# - containers
+# - wrapped `flox build` binaries.
+#
+# hook.on-activate could exit, in which case we want to cleanup this activation
+# rather than leaving it forever marked as ready = false
+if [ -n "${_FLOX_WATCHDOG_BIN:-}" ]; then
+  # TODO: Some of these args can be removed after https://github.com/flox/flox/issues/2206
+  "$_daemonize" \
+    -E _FLOX_WATCHDOG_LOG_LEVEL="debug" \
+    "$_FLOX_WATCHDOG_BIN" \
```

Comment:
should we add a `daemonize` flag to the watchdog instead?
Weak/mid opinion: Not sure whether that is useful or strictly better than this.

Reply:
I think this is good for now but something to keep in mind especially since we discussed starting the watchdog in Rust